Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM node:alpine3.10
+LABEL author "Wes Lambert, wlambertts@gmail.com"
+LABEL description="Dockerised version of Cyberchef server (https://github.com/gchq/CyberChef-server)"
+LABEL copyright "Crown Copyright 2020"
+LABEL license "Apache-2.0"
+RUN apk update && apk add --no-cache --update npm && \
+    npm cache clean --force && \
+    apk add --no-cache git && \
+    git clone https://github.com/gchq/CyberChef-server && \
```

Comment:
Seeing as the user has to clone this repo already to make this docker image, I would argue that we could just `COPY` the application directory into the Dockerfile context instead of cloning it again. Something like:

`COPY . /CyberChef-server`

(I haven't checked this)

Reply:
The intention with the Docker image was to make it to where you didn't need to have to deal with npm or anything like that.  I feel like this sort of moves away from that

I know very little about node/npm itself, but the following seems to work okay:

```
FROM node:alpine3.10
LABEL author "Wes Lambert, wlambertts@gmail.com"
LABEL description="Dockerised version of Cyberchef server (https://github.com/gchq/CyberChef-server)"
LABEL copyright "Crown Copyright 2020"
LABEL license "Apache-2.0"
COPY . /CyberChef-server
RUN apk update && apk add --update npm &&\
    npm cache clean --force && \
    npm install /CyberChef-server
ENTRYPOINT ["/usr/bin/npm", "--prefix", "/CyberChef-server", "run", "prod"]

```

Reply:
I think we could remove the apk install of `npm` (as it already exists) and explicitly move into the right directory, it could work:
```
FROM node:alpine3.10
LABEL author "Wes Lambert, wlambertts@gmail.com"
LABEL description="Dockerised version of Cyberchef server (https://github.com/gchq/CyberChef-server)"
LABEL copyright "Crown Copyright 2020"
LABEL license "Apache-2.0"
COPY . /CyberChef-server
RUN npm cache clean --force && \
         npm install /CyberChef-server
ENTRYPOINT ["/usr/bin/npm", "--prefix", "/CyberChef-server", "run", "prod"]

```

Reply:
Yeah, what I meant was that if I ran it like the above (after removing the install/upgrade), it would not find `npm`.  However, if I switch that to the following (switching from `/usr/bin/npm` to just `npm`), it works:

```
FROM node:alpine3.10
LABEL author "Wes Lambert, wlambertts@gmail.com"
LABEL description="Dockerised version of Cyberchef server (https://github.com/gchq/CyberChef-server)"
LABEL copyright "Crown Copyright 2020"
LABEL license "Apache-2.0"
COPY . /CyberChef-server
RUN npm cache clean --force && \
         npm install /CyberChef-server
ENTRYPOINT ["npm", "--prefix", "/CyberChef-server", "run", "prod"]
```